src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1446,6 +1431,25 @@ export class SavedObjectsRepository {
     const { refresh = DEFAULT_REFRESH_SETTING } = options;
     const savedObjectsBulkResponse = await this.bulkGet(savedObjects);
 
+    const errorObject = savedObjectsBulkResponse.saved_objects.find((obj) => !!obj.error);
+    if (errorObject) {
+      throw SavedObjectsErrorHelpers.decorateBadRequestError(new Error(errorObject.error?.message));
+    }
```

Comment:
I think it would be better that we throw the error with all the errorObjects.

Reply:
updated